### PR TITLE
ST GoBGP backend

### PR DESCRIPTION
**Pull request by ishidawataru**
test GoBGP backend as well as BIRD.

PRs below needs to be merged before this PR.

https://github.com/projectcalico/libcalico/pull/173
https://github.com/projectcalico/calico-bgp-daemon/pull/8

### CODE REVIEW

**Comment by tomdee** [comment]
Blocked on calico-bgp-daemon PR

**Comment by marvin-tigera** [comment]
Can one of the admins verify this patch?

File: tests/st/bgp/test_single_route_reflector.py [hunk 1]
```diff
@@ -62,3 +68,11 @@ def test_single_route_reflector(self):
                                                        workload_host2],
                                         ip_pass_list=[workload_host1.ip,
                                                       workload_host2.ip])
+
+    @attr('slow')
+    def test_bird_route_reflector_cluster(self):
```
Comment (robbrockbank) [inline tests/st/bgp/test_single_route_reflector.py]: _cluster should probably just be _single

File: tests/st/bgp/test_single_route_reflector.py [hunk 2]
```diff
@@ -62,3 +68,11 @@ def test_single_route_reflector(self):
...
+        self._test_single_route_reflector(backend='bird')
+
+    @attr('slow')
+    def test_gobgp_route_reflector_cluster(self):
```
Comment (robbrockbank) [inline tests/st/bgp/test_single_route_reflector.py]: ditto

File: tests/st/bgp/test_global_peers.py [hunk 3]
```diff
@@ -40,8 +39,8 @@ def test_global_peers(self):
                         additional_docker_options=ADDITIONAL_DOCKER_OPTIONS,
```
Comment (robbrockbank) [inline tests/st/bgp/test_global_peers.py]: The ADDITIONAL_DOCKER_OPTIONS have been made part of the common code and removed from bgp.

Can now use CLUSTER_STORE_DOCKER_OPTIONS in docker_host.py

**Comment by ishidawataru** [comment]
Reflected review comments and rebased on the master.

https://github.com/projectcalico/calico-bgp-daemon/pull/20 is needed to be merged to pass ST.

**Pull request closed without merging**